cupyx/scipy/sparse/dia.py

```diff
@@ -33,7 +33,13 @@ class dia_matrix(data._data_matrix):
     format = 'dia'
 
     def __init__(self, arg1, shape=None, dtype=None, copy=False):
-        if isinstance(arg1, tuple):
+        if _scipy_available and scipy.sparse.issparse(arg1):
```

Comment:
Shouldn't check that `arg1` is also stored in dia format?

Reply:
Good point! Thanks 😄

Should be fixed in the latest commit 🙂